### ISSUE TITLE
spipe doesn't gracefully handle an initial connection failure

### ISSUE THREAD

**Issue by gperciva** [Tarsnap/spiped#270]
`spipe` appears to hang when it's pointed at an unreachable socket:
```
$ ./spipe/spipe -t /bad-sock-name -k /dev/null
spipe: Could not connect
```

(the same applies for an IP address that isn't in use, such as `192.168.2.123:8001` on my local network)

It's easy to cancel it with `^C`, but it would be nicer if it could quit by itself (with exit code 1, rather than 130 for the `SIGINT` I sent it manualy).

**Comment by cperciva** [comment]
Oops.  It looks like we catch the error as soon as we try to write to the socket we failed to open though:

```
$ spipe -t /nosuchsock -k /COPYRIGHT 
spipe: Could not connect
asdad
spipe: Error writing: Bad file descriptor
```

Clearly something's broken in the error-handling path...

**Comment by gperciva** [comment]
Is the "success instead of ESRC" [1] included in 12.1-RELEASE-p8, or is that only going into 13.0?

[1] https://svnweb.freebsd.org/base?view=revision&revision=357894

**Comment by gperciva** [comment]
Never mind the `ESRCH` question.

**Comment by gperciva** [comment]
Fixed in #274.
